clang-tools-extra/clang-tidy/portability/AvoidPragmaOnceCheck.cpp

```diff
@@ -0,0 +1,50 @@
+//===--- AvoidPragmaOnceCheck.cpp - clang-tidy ----------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "AvoidPragmaOnceCheck.h"
+
+#include "clang/Basic/SourceManager.h"
+#include "clang/Lex/PPCallbacks.h"
+#include "clang/Lex/Preprocessor.h"
+#include "llvm/ADT/StringRef.h"
+
+namespace clang::tidy::portability {
+
+class PragmaOnceCallbacks : public PPCallbacks {
+public:
+  PragmaOnceCallbacks(AvoidPragmaOnceCheck *Check, const SourceManager &SM)
+      : Check(Check), SM(SM) {}
+  void PragmaDirective(SourceLocation Loc,
+                       PragmaIntroducerKind Introducer) override {
+    auto Str = llvm::StringRef(SM.getCharacterData(Loc));
+    if (!Str.consume_front("#")) {
+      return;
+    }
```

Comment:
```suggestion
    if (!Str.consume_front("#"))
      return;
```
Avoid writing braces for single `if`, `for`, etc..
Full list of style issues can be found in https://llvm.org/docs/CodingStandards.html#style-issues

Reply:
Thanks, do we have the `.clang-tidy` configuration  for the LLVM styles?

Reply:
There is a `.clang-tidy` in the root of LLVM repo that cover some of the style.